### ISSUE TITLE
Create a Dockerfile for running Cook in its own container

### ISSUE THREAD

**Issue by joshuastorck** [twosigma/Cook#83]
I tried creating my own Dockerfile and running Cook in a container, but it did not connect to the Mesos scheduler. I was using Docker 1.8.3 and using the --link option to docker run to make the zookeeper Docker container available to the Cook container.

**Comment by dgrnbrg** [comment]
Did you have a problem running Cook w/o a Dockerfile? Perhaps I can help you start by getting that working.

**Comment by joshuastorck** [comment]
Correct, I couldn't get it to connect to the mesos master when running in a
container, even with the docker bridge IP addresses. However, I could get
PySpark running in a separate container connect to the mesos master in a
configuration without the cook scheduler.

I'd be happy to sit down with you and show you how I was running it.

**Comment by dgrnbrg** [comment]
Could you get cook to connect without docker in the mix?

**Comment by joshuastorck** [comment]
I didn't try that since I wasn't trying to test it outside of the container.

**Comment by davvdg** [comment]
Hi, I don't know if you managed to solve your problem, but I've made a docker container to run cook (https://hub.docker.com/r/davvdg/docker-cook/), which I run through Marathon.

Mesos was not happy with it until I've configured it as net HOST rather than net BRIDGE. Actually, all my other framework in container (like spark, or airflow) need to be run as HOST for the registration process to occure. I've been able to submit jobs and witnesses that cook tried to execute them as mesos task, unfortunatly they all failed and I still need to figure out why.

Anyway, net HOST was the way to go for me.
